trigger.js

```diff
@@ -105,10 +108,30 @@ class TriggerHappy {
         this._executeTriggers(triggers);
     }
 
-    _onUpdateJournal(journal) {
-        if (journal.id === this._journalId || journal.name === this.journalName)
+    _onUpdateJournal(journal, update) {
+        if (update._id === this._journalId || journal.name === this.journalName)
             this._parseJournal();
     }
+
+    _onEndUpdate(scene, userId, update) {
+        if (update["x"] === undefined && update["y"] === undefined) return true;
+        let sceneTokens = scene.data.tokens;
+        let token = sceneTokens.find(t=>t._id === update._id);
+        if (token.hidden) return true; // hidden tokens don't trigger the trigger
+        let tokenNames = sceneTokens.map(t=>t.name); 
+        const triggers = this.triggers.filter(trigger => {
+            if (trigger.trigger.constructor.name !== "Token" || !tokenNames.includes(trigger.trigger.data.name) || trigger.trigger.data.name === token.name)

```

Comment:
YYou're specifically ignoring all triggers that aren't of type Token (what about Actor triggers?) and the token triggers that aren't in the scene. I don't think it's worth doing that.. my brain hurt before I understood that line of logic. It's not that it's bad, but it was a bit confusing.
It's smart though to skip the trigger if it's its own trigger. I hadn't thought of that.

I feel like the logic here needs to be different though, not sure how....